### ISSUE TITLE
The problem with file path based hashes in CSS Modules

### ISSUE THREAD

**Issue by roman01la** [parcel-bundler/lightningcss#420]
Hey. I just learnt that hash is coming from an absolute file path, is that correct?

It's problematic in a way that compiling the same module from a different location emits two different outputs. That's a valid case when multiple build tools are running from different directories, but the output js code still depends on the same CSS bundle.

If lightningcss would compute hash from file contents, it would be safe to run the build from any location on the file system. Also a useful side effect of this approach is deduplicated styles, for those rare cases when there are multiple identical modules, but perhaps it would make more sense than to hash individual CSS blocks, anyway.

That said, is there a way to configure hashing?

**Comment by mischnic** [comment]
See https://github.com/parcel-bundler/lightningcss/issues/355, you can set a root directory and then only the relative path to the file will be hashed.

**Comment by roman01la** [comment]
> See the commit linked in #355, you can set a root directory and then only the relative path to the file will be hashed.

Does that mean if I want absolutely consistent hashes I'd have to set the root to "/" basically?

**Comment by roman01la** [comment]
I guess the project root option is not exposed in a CLI?

**Comment by devongovett** [comment]
In the CLI, the project root is set to the current working directory

https://github.com/parcel-bundler/lightningcss/blob/366d923ef1f5789749d3acfb7014ab24694b8038/src/main.rs#L71

**Comment by roman01la** [comment]
yeah, I figured I have to use `transform` via Node API, but then the transform step becomes slower than just calling into the CLI directly
